flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/EmbeddedRocksDBStateBackendTest.java

```diff
@@ -619,6 +623,25 @@ public void testSharedIncrementalStateDeRegistration() throws Exception {
         }
     }
 
+    @Test(expected = FlinkRuntimeException.class)
+    public void testMapStateClear() throws Exception {
+        setupRocksKeyedStateBackend();
+        keyedStateBackend = spy(keyedStateBackend);
+        MapStateDescriptor<Integer, String> kvId =
+                new MapStateDescriptor<>("id", Integer.class, String.class);
+        MapState<Integer, String> state =
+                keyedStateBackend.getPartitionedState(
+                        VoidNamespace.INSTANCE, VoidNamespaceSerializer.INSTANCE, kvId);
+
+        doAnswer(
+                        invocationOnMock -> {
+                            throw new RocksDBException("Artificial failure");
+                        })
+                .when(keyedStateBackend)
+                .getWriteOptions();
```

Comment:
We could change the implementation of state#clear in the future, and we might not call `keyedStateBackend`, however, we would not change the kernel logic of mapState#clear that we would always create a rocksdb iterator.

I think the logic of spying on creating a iterator could be referred:
~~~ java
        doAnswer(
                        new Answer<Object>() {

                            @Override
                            public Object answer(InvocationOnMock invocationOnMock)
                                    throws Throwable {
                                RocksIterator rocksIterator =
                                        spy((RocksIterator) invocationOnMock.callRealMethod());
                                allCreatedCloseables.add(rocksIterator);
                                return rocksIterator;
                            }
                        })
                .when(keyedStateBackend.db)
                .newIterator(any(ColumnFamilyHandle.class), any(ReadOptions.class));
~~~

Reply:
I agree. I have mocked the `newIterator` to throw `RocksDBException`.